src/nwbinspector/checks/time_series.py

```diff
@@ -129,3 +129,14 @@ def check_resolution(time_series: TimeSeries):
         return InspectorMessage(
             message=f"'resolution' should use -1.0 or NaN for unknown instead of {time_series.resolution}."
         )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=TimeSeries)
+def check_rate_is_not_zero(time_series: TimeSeries):
+    if time_series.data is not None:
+        data_shape = get_data_shape(time_series.data) 
+    if time_series.timestamps is not None:
+        timestamps_shape = get_data_shape(time_series.timestamps)
+    if time_series.rate == 0.0:
+        if (data_shape[0] > 1 or timestamps_shape[0] > 1):
```

Comment:
A good first opportunity here to introduce you to [Cyclomatic Complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) and our approaches to keeping it as low as possible

Python has short-circuit logic for conditional statements, so that if an earlier one fails, it never proceeds to the next ones after

```python
    if time_series.rate == 0.0 and (data_shape[0] > 1 or timestamps_shape[0] > 1):
```

voila, the code is now nested by one less level. We try to keep code as flat as possible within reason (sometimes it's unavoidable of course...)

A lot of other inspirations we use for code style relate to [PEP 20](https://peps.python.org/pep-0020/), so I'd give that a quick look as well

Reply:
Thank you for this correction as well, I will keep this in mind